uaa/src/test/java/org/cloudfoundry/identity/uaa/orchestrator/OrchestratorClaimAT.java

```diff
@@ -1,124 +1,280 @@
+            getZone = OBJECT_MAPPER.readValue(getResponse, OrchestratorZoneResponse.class);
+
+            //assertions from get API
+            assertNotNull(getResponse);
+            assertNotNull(getZone);
+            assertEquals(zoneName, getZone.getName());
+            assertNotNull(getZone.getConnectionDetails().getDashboardUri());
+            assertNotNull(getZone.getConnectionDetails().getUri());
+            assertNotNull(getZone.getConnectionDetails().getIssuerId());
+            assertNotNull(getZone.getConnectionDetails().getZone().getHttpHeaderValue());
+            assertNotNull(getZone.getConnectionDetails().getZone().getHttpHeaderName());
+            if (uaaRequest.getSubdomain() != null) {
+                assertEquals(uaaRequest.getSubdomain(), getZone.getConnectionDetails().getSubdomain());
+            } else {
+                assertNotNull(getZone.getConnectionDetails().getSubdomain());
+            }
+            JsonNode outputSecret = orchestratorUtil.readSecret(claimName, ServiceName.UAA);
+            String outputSecretData = StringEscapeUtils.unescapeJava(outputSecret.toString());
+            assertNotNull(outputSecret);
+            assertNotNull(outputSecretData);
+            JsonNode connection = outputSecret.get("connectionDetails");
+            JsonNode connectionSecret = OBJECT_MAPPER.readTree(connection.asText());
+
+            // if claim and outputSecret(created by Orchestrator as result of successful instance creation) both are not null
+            // that means uaa zone is successfully created in uaa service.
+            UaaForProvider uaaforProvider = ((UaaClaimSpec) claim.getSpec()).getUaaForProvider();
+
+            //assertions from Claim.
+            assertNotNull(connection);
+            if (uaaRequest.getSubdomain() != null) {
+                assertEquals(uaaRequest.getSubdomain(), uaaforProvider.getSubdomain());
+            } else {
+                assertNotNull(connectionSecret.get("subdomain"));
+            }
+            assertEquals(getZone.getConnectionDetails().getUri(), connectionSecret.get("uri").asText());
+            assertEquals(getZone.getConnectionDetails().getDashboardUri(), connectionSecret.get("dashboardUrl").asText());
+            assertEquals(getZone.getConnectionDetails().getIssuerId(), connectionSecret.get("issuerId").asText());
+            assertEquals(getZone.getConnectionDetails().getZone().getHttpHeaderName(), connectionSecret.get("zone").get("http-header-name").asText());
+            assertEquals(getZone.getConnectionDetails().getZone().getHttpHeaderValue(), connectionSecret.get("zone").get("http-header-value").asText());
+        } finally {
+            if (isClaimCreated) {
+                deleteZone(claimName, ServiceName.UAA);
+            }
+            if (getZone != null) {
+                Thread.sleep(20000);
+                getZoneByName(zoneName, HttpStatus.NOT_FOUND, "Zone[" + namespace + "-" + claimName + "] not found.");
```

Comment:
you can replace `namespace + "-" + claimName` with zoneName

Reply:
Done